documentation/nat-service/nat-service.md

```diff
@@ -0,0 +1,225 @@
+# NAT scenario
+
+## Overview
+
+NAT scenario decribes 2 NAT use cases:
+
+* VMs in VNET access Internet, aka SNAT.
+* Users from Internet access servcies hosted on the cloud VMs in VNET, aka DNAT.
+
+We use network appliance in this HLD, acting the role of NAT gateway as well as underlay router.
+This scenario provides the following capabilites:
+
+* SNAT service for TCP, UDP and ICMP packets.
+* DNAT service for TCP, UDP and ICMP packets.
+* Routing service for underlay forwarding.
+* Billing service(by traffic volumn and by commited traffic rate).
+
+The goal is to test the following properties:
```

Comment:
This seems a little bit out of place, describing an experiment. If this document is an HLD, it should simply be a description and specification of something and not talk about what-if scenarios you wish to test.